chat-server.py

```diff
@@ -0,0 +1,84 @@
+import socket
+import selectors
+
+HEADER_LENGTH = 10
+HOST = '127.0.0.1'
+PORT = 54321
+clients = []  # List for storing client information
+
+
+def accept_connection(sock):
+	"""Function to accept a new connection"""
+
+	global clients
+	conn, addr = sock.accept()
+	print(f'Connected to by {addr}')
+	conn.setblocking(False)
+
+	# Receive first message. This will contain username
+	user_header = conn.recv(HEADER_LENGTH)
+	username_length = int(user_header.decode("utf-8").strip())
```

Comment:
Rather than do this server side I've now validated username and message input client side to check for `UnicodeEncodeError`. Does it reason that if a `UnicodeEncodeError` isn't thrown client side then a `UnicodeDecodeError` won't be thrown server side?

Reply:
That holds true as long as you trust the client. If a miscreant modified your client script, they could still crash the server.

Reply:
Server now checks for `UnicodeDecodeError` when receiving usernames/messages